packages/framework/src/Component/Setting/SettingValue.php

```diff
@@ -89,6 +89,11 @@ public function getName()
      */
     public function getValue()
     {
+        if ($this->value === null && $this->type !== self::TYPE_NULL) {
+            $message = 'Correct setting value type for null value is "' . self::TYPE_NULL . '".';
```

Comment:
I would change this message to: Setting value type "' . $this->type . "' does not allow null value.